File: Examples/Rust/src/examples/get_port_names.rs
use ximu3::port_scanner::*;

pub fn run() {
    let port_names = PortScanner::get_port_names();

    if port_names.is_empty() {
        println!("No ports available")
    }

    for port_name in port_names {
        println!("{}", port_name);
    }
}

File: Examples/C/examples/get_port_names.c
#include "../helpers.h"
#include <stdio.h>
#include "Ximu3.h"

void get_port_names()
{
    const XIMU3_CharArrays port_names = XIMU3_port_scanner_get_port_names();

    if (port_names.length == 0)
    {
        printf("No ports available\n");
        return;
    }

    for (uint32_t index = 0; index < port_names.length; index++)
    {
        printf("%s\n", port_names.array[index]);
    }

    XIMU3_char_arrays_free(port_names);
}
